### ISSUE TITLE
Block screenshotting

### ISSUE THREAD

**Issue by daira** [Electric-Coin-Company/zashi-ios#1050]
If we want the user to be able to intentionally capture the seed phrase for example, that should be an explicit action that saves a file in the right place with the right permissions.

**Comment by true-jared** [comment]
@LukasKorba Is this even possible on iOS? In case yes, what would the user journey be here? We block them from using a functionality they are used to - so we have to handle it somehow, right? What would the flow be - displaying a warning modal when they attempt to do so? @daira

**Comment by LukasKorba** [comment]
> @LukasKorba Is this even possible on iOS? In case yes, what would the user journey be here? We block them from using a functionality they are used to - so we have to handle it somehow, right? What would the flow be - displaying a warning modal when they attempt to do so? @daira

Preventing screen capture in an iOS app is not directly supported by the iOS operating system. Apple’s design philosophy emphasises the user’s control over their own device, so preventing screen capture goes against those principles. 

However, we can’t restrict from capturing screenshot or recording video but we can hide our content. It's been a whole I did that and it was in older iOS and UIKit so this would need to be investigated if SwiftUI and newer iOS still supports that. Essentially how it works is the sensitive fields are marked as secure and when screenshot or recording is happening, those fields are hidden. So the screenshot is taken but all marked content is missing on it. 

It's a great solution because we don't block users from tsaking the screenshot but we don't let them to see the sensitive data. I think there was even possible to catch that action and show an alert like `we detected a screenshot on this screen but it holds very secure and sensitive data so we purposely hid it`. Let me know if that would work and also the priority, like I said, this will take some time to investigate. Also let me know if hiding the content is enough or detection code + some alert/notification would be part of this task.

**Comment by true-jared** [comment]
Agreed with Josh that this is outside of the scope for 1.0. It should be a part of the overall product strategy and roadmap planning, and there should be discussions around the user journey + experience, and ways of protecting the seed phrase in general. We want to do this right and not rush it for 1.0.
